pkg/secrets/exec_windows.go

```diff
@@ -0,0 +1,120 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018-present Datadog, Inc.
+
+//go:build secrets && windows
+// +build secrets,windows
+
+package secrets
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"syscall"
+
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/svc/mgr"
+
+	"github.com/DataDog/datadog-agent/pkg/util/winutil"
+)
+
+// commandContext sets up an exec.Cmd for running with a context
+func commandContext(ctx context.Context, name string, arg ...string) (*exec.Cmd, func(), error) {
+	cmd := exec.CommandContext(ctx, name, arg...)
+	done := func() {}
+	localSystem, err := getLocalSystemSID()
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not query Local System SID: %s", err)
+	}
+	defer windows.FreeSid(localSystem)
+
+	currentUser, err := winutil.GetSidFromUser()
+	if err != nil {
+		return nil, nil, fmt.Errorf("could not get SID for current user: %s", err)
+	}
+
+	// If we are running as Local System we need to "sandbox" execution to "ddagentuser"
+	if currentUser.Equals(localSystem) {
+		// Retrieve token from the running Datadog Agent service
+		token, err := getDDAgentServiceToken()
+		if err != nil {
+			return nil, nil, err
+		}
+
+		done = func() {
+			defer windows.CloseHandle(windows.Handle(token))
+		}
+
+		// Configure the token to run with
+		cmd.SysProcAttr = &syscall.SysProcAttr{Token: syscall.Token(token)}
+	}
+
+	return cmd, done, nil
+}
+
+// getDDAgentServiceToken retrieves token from the running Datadog Agent service
+func getDDAgentServiceToken() (windows.Token, error) {
+	var token, duplicatedToken windows.Token
+	pid, err := getServicePid("datadogagent")
```

Comment:
nit: This should probably be a constant up top. If you change that, we should update the formatted errors below.